liblttng-ust-java-agent/java/lttng-ust-agent-jul/org/lttng/ust/agent/jul/LttngLogHandler.java

```diff
@@ -134,13 +158,33 @@ public void publish(LogRecord record) {
 		 */
 		LttngJulApi.tracepointWithContext(formattedMessage,
 				record.getLoggerName(),
-				record.getSourceClassName(),
-				record.getSourceMethodName(),
+				logSource ? record.getSourceClassName() : EMPTY,
+				logSource ? record.getSourceMethodName() : EMPTY,
 				record.getMillis(),
 				record.getLevel().intValue(),
 				record.getThreadID(),
 				contextInfo.getEntriesArray(),
 				contextInfo.getStringsArray());
 	}
 
+	/**
+	 * Set whether the source method/class should be logged with the events.
+	 * Computing the source has a non-negligible overhead. By default, those
+	 * fields are logged and need to be explicily disabled.
+	 *
+	 * @param doLog
+	 *            <code>true</code> if the source method/class should be logged
+	 *            with the events, <code>false</code> otherwise.
+	 */
+	public void setLogSource(boolean doLog) {
+		logSource = doLog;
+	}
+	
+	/**
+	 * Get whether the source method/class should be logged with the events.
+	 */
+	public boolean getLogSource() {
```

Comment:
Technically getters for booleans are called `is*` (`isLogSource()`), but meh not a big deal!